Sources/FluentUI_iOS/Components/Pill Button Bar/SwiftUI/PillButtonBarView.swift

```diff
@@ -0,0 +1,235 @@
+                if let value = newValue {
+                    selected.wrappedValue = value
+                }
+            }
+        )
+    }
+
+    public var body: some View {
+        ScrollViewReader { scrollProxy in
+            ScrollView(.horizontal, showsIndicators: false) {
+                pillButtonStack()
+                    .environment(\.isEnabled, isEnabled)
+            }
+            .environment(\.isEnabled, true)
+            .scrollDisabled(disableScroll)
+            .onGeometryChange(for: CGRect.self) { proxy in
+                proxy.frame(in: .global)
+            } action: { newFrame in
+                scrollViewFrame = newFrame
+                containerWidth = newFrame.width
+            }
+            .onAppear {
+                DispatchQueue.main.async {
+                    scrollToSelectedPill(scrollProxy)
+                }
+            }
+            .onChange_iOS17(of: selected) { _ in
+                guard !disableScroll,
+                      let selected = currentSelection.wrappedValue,
+                      let pillFrame = pillFrames[selected] else {
+                    return
+                }
+
+                let scrollMinX = scrollViewFrame.minX
+                let scrollMaxX = scrollViewFrame.maxX
+
+                if pillFrame.maxX > scrollMaxX {
+                    withAnimation {
+                        scrollProxy.scrollTo(selected,
+                                             anchor: UnitPoint(x: Constants.scrollAnchorTrailingX - (Constants.scrollPadding / max(scrollViewFrame.width, Constants.scrollAnchorTrailingX)),
+                                                               y: Constants.scrollAnchorY))
+                    }
+                } else if pillFrame.minX < scrollMinX {
+                    withAnimation {
+                        scrollProxy.scrollTo(selected,
+                                             anchor: UnitPoint(x: Constants.scrollPadding / max(scrollViewFrame.width, Constants.scrollAnchorTrailingX),
+                                                               y: Constants.scrollAnchorY))
+                    }
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func pillButtonStack() -> some View {
+        HStack(spacing: Constants.pillSpacing) {
+            ForEach(viewModels, id: \.id) { item in
+                pillButton(for: item)
+            }
+        }
+        .onGeometryChange(for: CGFloat.self) { proxy in
+            proxy.size.width
+        } action: { newWidth in
+            contentWidth = newWidth
+        }
+        .padding(.vertical, Constants.verticalPadding)
+        .padding(.horizontal, Constants.horizontalPadding)
+        .frame(minWidth: disableScroll ? max(0, containerWidth - Constants.horizontalPadding * 2) : 0,
+               alignment: shouldCenterAlign ? .center : .leading)
+    }
+
+    private func pillButton(for item: PillButtonViewModel<Selection>) -> some View {
```

Comment:
nit: since we're passing a view model should we just rename `item` to `viewModel`